app/src/lib/popup-manager.ts

```diff
@@ -47,8 +47,19 @@ export class PopupManager {
    *  otherwise returns the first non-error type popup.
    */
   public get currentPopup(): Popup | null {
+    return this.allPopups.at(-1) ?? null
+  }
+
+  /**
+   * Returns all the popups in the stack. If there are error popups, it returns
+   *  them on the top of the stack (the end of the array -> last on, last off).
+   */
+  public get allPopups(): ReadonlyArray<Popup> {
     const errorPopups = this.getPopupsOfType(PopupType.Error)
-    return errorPopups.at(-1) ?? this.popupStack.at(-1) ?? null
+    const nonErrorPopups = this.popupStack.filter(
+      p => p.type !== PopupType.Error
+    )
+    return [...nonErrorPopups, ...errorPopups]
```

Comment:
Given that this getter is called for each AppStore emit I'd rather we avoided doing array mutation here if we can avoid it, lots of unnecessary GC pressure. I'm not saying we need to fix this in this PR. If you have time I promise I'll re-review it tomorrow morning but if not it'd be great if we could put this on a TODO list somewhere and change this such that we constructs allPopups when the underlying array mutates.